libraries/DateTime/contracts/DateTime.aes

```diff
@@ -0,0 +1,244 @@
+contract DateTime =
+  /*
+   Date and Time utilities for sophia contracts
+  */
+
+// Define used structures
+  record state = 
+    {
+      origin_year : int, 
+      day_in_seconds : int,
+      year_in_seconds : int,
+      leap_year_in_seconds : int,
+      hour_in_seconds : int,
+      minute_in_seconds : int
+    }
+
+  record date_time = 
+    {
+      year : int,
+      month : int,
+      day : int,
+      hour : int,
+      minute : int,
+      second : int,
+      weekday : int,
+      timestamp : int
+    }
+  
+  record buff = 
+    {
+      year : int,
+      month : int,
+      day : int,
+      leap_years_before : int,
+      seconds_accounted_for : int,
+      timestamp : int
+    }                    
+
+// Public function (API)
+  public function init() : state =
+    {
+      origin_year = 1970, 
+      day_in_seconds = 86400,
+      year_in_seconds = 31536000,
+      leap_year_in_seconds = 31622400,
+      hour_in_seconds = 3600,
+      minute_in_seconds = 60
+      }
+
+  public function parse_timestamp(timestamp : int) : date_time = 
+    let dt : date_time = init_date_time(timestamp)
```

Comment:
maybe this function is useless.. you set timestamp in 'init_date_time'

Reply:
Could you clarify please?

Reply:
in init_Date_time you return this record

Reply:
Ok, thank you. `Buff` struct is now removed.